src/AutoRest.CSharp/Common/Generation/Writers/ClientOptionsWriter.cs

```diff
@@ -50,6 +51,21 @@ public static void WriteClientOptions(CodeWriter writer, ClientOptionsTypeProvid
                             }
                         }
                     }
+
+                    if (clientOptions.AdditionalProperties != null && !Configuration.KeepOptionalClientParametersInConstructor)
+                    {
+                        if (clientOptions.ApiVersions is not null)
+                        {
+                            writer.Line();
+                        }
+
+                        foreach (var property in clientOptions.AdditionalProperties)
+                        {
+                            writer.WriteXmlDocumentationSummary(property.Description);
+                            writer.Line($"public {property.Type} {property.Name.FirstCharToUpperCase()} {{ get; set; }}");
```

Comment:
Shall we use other normalization method like `property.Name.ToCleanName()`?

Reply:
Not sure, `ToCleanName` is to complicated, I'm not sure if there is any side effects. So I actually prefer to use `FirstCharToUpperCase` first and we could change later is there are some special cases.

Reply:
I don't have strong opinion. If you insist, I could change.

Reply:
In `ModelWriter`, we use this: https://github.com/Azure/autorest.csharp/blob/d472a8294859384089e456a8d8d80a18e328a479/src/AutoRest.CSharp/Common/Generation/Writers/ModelWriter.cs#L104

It's from `Declaration.ActualName`: https://github.com/Azure/autorest.csharp/blob/d472a8294859384089e456a8d8d80a18e328a479/src/AutoRest.CSharp/Common/Output/Models/FieldDeclaration.cs#L14

which comes from `CodeWriterDeclaration.SetActualName`: https://github.com/Azure/autorest.csharp/blob/d472a8294859384089e456a8d8d80a18e328a479/src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterDeclaration.cs#L22

There are a few ways to set `ActualName`. One path comes from `ToCleanName()`:

https://github.com/Azure/autorest.csharp/blob/d472a8294859384089e456a8d8d80a18e328a479/src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProviderFields.cs#L57

https://github.com/Azure/autorest.csharp/blob/d472a8294859384089e456a8d8d80a18e328a479/src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProviderFields.cs#L44